src/async_impl/request.rs

```diff
@@ -145,7 +145,18 @@ impl RequestBuilder {
     }
 
     /// Add a `Header` to this Request.
-    pub fn header<K, V>(mut self, key: K, value: V) -> RequestBuilder
+    pub fn header<K, V>(self, key: K, value: V) -> RequestBuilder
+    where
+        HeaderName: TryFrom<K>,
+        <HeaderName as TryFrom<K>>::Error: Into<http::Error>,
+        HeaderValue: TryFrom<V>,
+        <HeaderValue as TryFrom<V>>::Error: Into<http::Error>, 
+    {
+        self.header_sensitive(key, value, false)
+    }
+
+    /// Add a `Header` to this Request with ability to define if header_value is sensitive.
+    pub fn header_sensitive<K, V>(mut self, key: K, value: V, sensitive: bool) -> RequestBuilder
```

Comment:
For now, I'd keep this method private.

Reply:
Ok, I will do the change.
In the future, it could be interesting to add custom sensitive header.

But one step at a time :)

Reply:
To be fair, a user can create a `HeaderValue` themselves and set the flag, pass *that* to `.header(name, my_val)`.

Reply:
Yes, that's what I did before opening this Pull Request.
Thanks for the merge